app/src/main/java/ipvc/gymbuddy/app/datastore/ContractDataStore.kt

```diff
@@ -0,0 +1,44 @@
+package ipvc.gymbuddy.app.datastore
+
+import android.annotation.SuppressLint
+import android.content.Context
+import androidx.lifecycle.MutableLiveData
+import ipvc.gymbuddy.api.core.RequestResult
+import ipvc.gymbuddy.api.models.Contract
+import ipvc.gymbuddy.api.services.ContractService
+import ipvc.gymbuddy.app.core.AsyncData
+import kotlinx.coroutines.launch
+
+class ContractDataStore(context: Context) : BaseDataStore(context) {
+
+    @SuppressLint("StaticFieldLeak")
+    companion object{
+        @Volatile private var instance: ContractDataStore? = null
+
+        fun getInstance(context: Context) = instance ?: synchronized(this) {
+            instance ?: ContractDataStore(context).also { instance = it }
+        }
+    }
+
+    private val authenticationDataStore = AuthenticationDataStore.getInstance(context)
+    var contracts = MutableLiveData<AsyncData<List<Contract>>>(AsyncData(listOf()))
+
+
+    fun getContracts() {
+        val user = authenticationDataStore.user.value
+        contracts.postValue(AsyncData(contracts.value?.data, AsyncData.Status.LOADING))
+        coroutine.launch {
+            if (user != null) {
```

Comment:
if user is null then it should not even postvalue to loading status and trigger the coroutine..